### PR TITLE
#32 expose statistics

### PR DESCRIPTION
**Pull request by RaidenRabit**

### CODE REVIEW

File: Code/Database/InsertScript.sql [hunk 1]
```diff
@@ -117,31 +117,33 @@ INSERT INTO Copy (CopyID, ISBN, TypeName, LibraryName)
...
 		(17,7,'needed books','Pikalo');
 SET IDENTITY_INSERT Copy OFF;
 
 INSERT INTO Loan (CopyID, SSN, FromDate, ToDate)
 VALUES (5,123456786,GETDATE(),null),
-		(2,123456786,GETDATE(),GETDATE()),
+		(2,123456786,GETDATE(),GETDATE() + 1),
 		(6,123456789,GETDATE(),null),
-		(1,123456789,GETDATE(),GETDATE()),
+		(1,123456789,GETDATE(),GETDATE() + 2),
```
Comment (bubriks) [inline Code/Database/InsertScript.sql]: what is up with these + x number?

File: Code/Database/SQLCreateQuery.sql [hunk 2]
```diff
@@ -98,6 +98,14 @@ CREATE TABLE Loan (
 );
 GO
 
+CREATE NONCLUSTERED INDEX ix_CopyIsbn_ReferenceMaterialIsbn
+ON [dbo].[Copy]
```
Comment (bubriks) [inline Code/Database/SQLCreateQuery.sql]: why do you use these brackets we didnt use them in the prev code

File: Code/Database/SQLCreateQuery.sql [hunk 3]
```diff
@@ -237,6 +245,29 @@ BEGIN
...
+	SELECT FromDate, ToDate FROM Loan where ToDate is NOT null
+	  )
+
+	SELECT CAST((SELECT SUM(DATEDIFF(DAY, FromDate, ToDate)) FROM a)/(SELECT COUNT(*) FROM a) AS DECIMAL(10, 2)) AS AverageLoaningTime
+END
+GO
+
+DROP PROCEDURE IF EXISTS TopLoaningLibrary
+GO
+CREATE PROCEDURE TopLoaningLibrary
```
Comment (bubriks) [inline Code/Database/SQLCreateQuery.sql:263]: why are both of these procedures not views?

File: Code/GeorgiaLibrarySystem-/Tests/IntegrationTest/StatisticsTest.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            //Arrange
+            Setup(approach);
+
+            //Act
+            var result = _statisticsService.TopTenBooks();
+
+            //Assert
+            Assert.AreEqual(result.Count, totalCount);
+            Assert.AreEqual(result[0].loaned_count, loanedCopies, "in reality it was: " + result[0].loaned_count);
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/Tests/IntegrationTest/StatisticsTest.cs:28]: is this truly correct, I mean your checking 2 different counts in the same test

**Review by bubriks — COMMENTED** [review]
Overall seams nice and simple as the thing should be, but i would like some quick answers before i approve :D

File: Code/Database/InsertScript.sql (hunk 1 above)
Comment (RaidenRabit) [reply Code/Database/InsertScript.sql]: just making sure the avg loan time isnt 0.
this way itll be: loaned today, returned today + x. so the day after tomorrow :D

File: Code/Database/SQLCreateQuery.sql (hunk 2 above)
Comment (RaidenRabit) [reply Code/Database/SQLCreateQuery.sql]: oh, i was too lazy to type them so i just drag+drop the thing, ill remove it

File: Code/GeorgiaLibrarySystem-/Tests/IntegrationTest/StatisticsTest.cs (hunk 4 above)
Comment (RaidenRabit) [reply Code/GeorgiaLibrarySystem-/Tests/IntegrationTest/StatisticsTest.cs:28]: yes

File: Code/Database/SQLCreateQuery.sql (hunk 3 above)
Comment (RaidenRabit) [reply Code/Database/SQLCreateQuery.sql:263]: technically those are returning just 1 thing, as opposed to a list of things. other than that, i just wanted to see how to make procedures

File: Code/Database/InsertScript.sql (hunk 1 above)
Comment (bubriks) [reply Code/Database/InsertScript.sql]: ok i guess, but right now the return time will be some time in future :D

File: Code/Database/SQLCreateQuery.sql (hunk 3 above)
Comment (bubriks) [reply Code/Database/SQLCreateQuery.sql:263]: ok

**Review by bubriks — APPROVED** [review]

File: Code/Database/InsertScript.sql (hunk 1 above)
Comment (RaidenRabit) [reply Code/Database/InsertScript.sql]: i can just put - on the FromDate, if youd like it more

Comment (bubriks) [reply Code/Database/InsertScript.sql]: it doesnt matter in the end, its just seed

**Pull request merged**